### PR TITLE
Checks if tool is installed (svn, cvs, hg)

### PR DESCRIPTION
**Pull request by ignacio**
As #387, but for CVS, Mercurial and Subversion.

### CODE REVIEW

File: src/luarocks/fetch/cvs.lua [hunk 1]
```diff
@@ -20,9 +20,14 @@ function cvs.get_sources(rockspec, extract, dest_dir)
    assert(type(rockspec) == "table")
    assert(type(dest_dir) == "string" or not dest_dir)
 
+   local cvs_cmd = rockspec.variables.CVS
+   if not fs.execute_quiet(cvs_cmd, "--version") then
+      return nil, "'"..cvs_cmd.."' program not found. Is CVS installed? You may want to edit variables.CVS"
```
Comment (Tieske) [inline src/luarocks/fetch/cvs.lua]: wouldn't it be clearer to write; "You may want to edit the 'variables.CVS' value in file 'config.lua' "

Comment (hishamhm) [inline src/luarocks/fetch/cvs.lua]: alternatively, `"'"..cvs_cmd.."' program not found. Please make sure it is available in your PATH".` _might_ be an easier instruction to follow (not sure).

Comment (siffiejoe) [inline src/luarocks/fetch/cvs.lua]: It doesn't have to be in PATH if it's an absolute path.

Comment (ignacio) [inline src/luarocks/fetch/cvs.lua]: Sure. I copied this message from the cmake builder. Considering that:
1) The user just wants to install a rock. She might have no idea what "hg" is.
2) The tool's executable may not indicate the tool's name (hg -> Mercurial)

What about:
'foo' program not found. Make sure Bar is installed and is available in your PATH"

File: src/luarocks/fetch/hg.lua [hunk 2]
```diff
@@ -21,9 +21,13 @@ function hg.get_sources(rockspec, extract, dest_dir)
...
 
    local hg_cmd = rockspec.variables.HG
+   if not fs.execute_quiet(hg_cmd, "--version") then
+      return nil, "'"..hg_cmd.."' program not found. Is Mercurial installed? You may want to edit variables.HG"
+   end
+
    local name_version = rockspec.name .. "-" .. rockspec.version
    -- Strip off special hg:// protocol type
-   local url = rockspec.source.url:gsub("^hg://", "") 
+   local url = rockspec.source.url:gsub("^hg://", "")
```
Comment (ignacio) [inline src/luarocks/fetch/hg.lua]: Sorry @hishamhm that trailing whitespace trigger my OCD :smile:

Comment (hishamhm) [inline src/luarocks/fetch/hg.lua]: np :)

File: src/luarocks/fetch/cvs.lua (hunk 1 above)
Comment (hishamhm) [inline src/luarocks/fetch/cvs.lua]: > It doesn't have to be in PATH if it's an absolute path.

Sure, I was just catering for the common case. If git is not found, most likely it is simply not installed and installing via the OS's usual methods will put it there.

Comment (Tieske) [inline src/luarocks/fetch/cvs.lua]: not on Windows...

**Comment by ignacio** [comment]
So, do you people think my proposed wording is ok?

**Comment by hishamhm** [comment]
The refactored function is a good idea and I think the wording is good.

**Pull request merged**